sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1912,6 +1913,26 @@ class Analyzer(
           nondeterToAttr.get(e).map(_.toAttribute).getOrElse(e)
         }.copy(child = newChild)
 
+      case j: Join if j.condition.isDefined && !j.condition.get.deterministic =>
+        j match {
+          // We can push down non-deterministic joining keys.
```

Comment:
IIUC, we should not have the same concern (e.g., rand(1) > 0 && rand(11) < 0) on joining keys. The format of joining keys is "x equal to y". We will evaluate x and y for two joining tables. May you show an example we can't push down the joining keys?